src/avalanchemq/client/client.cr

```diff
@@ -231,6 +235,11 @@ module AvalancheMQ
     @write_lock = Mutex.new(:checked)
 
     def deliver(frame, msg)
+      return false if closed?
+      unless frame.channel.zero? || @channels[frame.channel]?.try &.running?
```

Comment:
I find it easier reading unless statements if they are only depending on one boolean value and not a mix.
```suggestion
      unless channel_closed?(frame)
```
``` crystal
     private def channel_closed?(frame)
        frame.channel.zero? || @channels[frame.channel]?.try &.running?
      end
```